packages/opencensus-core/src/stats/view.ts

```diff
@@ -294,7 +246,22 @@ export class BaseView implements View {
    * @param tags The desired data's tags
    * @returns {AggregationData}
    */
-  getSnapshot(tags: Tags): AggregationData {
-    return this.rows[this.encodeTags(tags)];
+  getSnapshot(tagValues: TagValue[]): AggregationData {
+    return this.tagValueAggregationMap[this.encodeTagValues(tagValues)];
+  }
+
+  /** Determines whether the given TagKeys are valid. */
+  private validateTagKeys(tagKeys: TagKey[]): TagKey[] {
+    const tagKeysCopy = Object.assign([], tagKeys);
```

Comment:
Why does this need to be copied?

Reply:
This is to make a shallow copy of tagKeys to protect internal functionality, in case any update on  original data.

Reply:
Oh, I see. I missed that `tagKeysCopy` is returned by the function.